core/src/syscall/write.rs

```diff
@@ -1,30 +1,37 @@
+use athena_interface::StatusCode;
+
 use crate::{
-  runtime::{Register, Syscall, SyscallContext},
+  runtime::{Outcome, Register, Syscall, SyscallContext, SyscallResult},
   utils::num_to_comma_separated,
 };
 
-pub struct SyscallWrite;
+/// Write bytes to selected file descriptor.
+/// Supported FDs:
+/// - 1: stdout
+/// - 2: stderr
+/// - 3: public values
+/// - 4: input stream (TODO:poszu check why it writes to the input stream)
+///
+/// FD 1 supports "cycle tracker". TODO(poszu): see if we need this, and add documentation or remove.
```

Comment:
re: cycle tracker, see https://docs.succinct.xyz/writing-programs/cycle-tracking.html. We can discuss whether we want to maintain this.

re: fd4 and writing to the input stream, I can't find any documentation on this in sp1. My guess is that it's useful for debugging and testing, and in theory could be used for cross-contract calls (inside the same runtime, which we don't support) or recursive proofs (which we don't support yet).

Reply:
Regarding the cycle tracker, I see a potential problem with it because it lets the contract code control some debugging/profiling functionality in the runtime. I'm unsure if we want to allow doing this by arbitrary programs. Perhaps this could be compiled in under a feature flag, only for builds instrumented for profiling and **not** enabled "in production".

Reply:
works for me. feel free to open an issue to follow up on this.